src/java/org/apache/cassandra/cql3/statements/SelectStatement.java

```diff
@@ -405,17 +405,20 @@ public ReadQuery getQuery(QueryOptions options,
                               long nowInSec,
                               DataLimits limit)
     {
-        boolean isPartitionRangeQuery = restrictions.isKeyRange() || restrictions.usesSecondaryIndexing();
+        RowFilter rowFilter = getRowFilter(options, state);
 
-        if (isPartitionRangeQuery)
+        if (restrictions.isKeyRange())
         {
-            if (restrictions.isKeyRange() && restrictions.usesSecondaryIndexing() && !SchemaConstants.isLocalSystemKeyspace(table.keyspace))
+            if (restrictions.usesSecondaryIndexing() && !SchemaConstants.isLocalSystemKeyspace(table.keyspace))
                 Guardrails.nonPartitionRestrictedIndexQueryEnabled.ensureEnabled(state);
```

Comment:
Just to confirm, this guard rail doesn't need to appear elsewhere since `isKeyRange` is the indicator of whether it is multiple partitions?

Reply:
Right, the idea is we trivially don't need to look at the guardrail if we're querying a single partition anyway.